lib/srv/desktop/rdp/rdpclient/src/client.rs

```diff
@@ -483,27 +487,36 @@ impl Client {
         Ok(())
     }
 
+    async fn handle_tdp_sd_info_response(
+        x224_processor: Arc<Mutex<X224Processor>>,
+        res: tdp::SharedDirectoryInfoResponse,
+    ) -> ClientResult<()> {
+        global::TOKIO_RT
+            .spawn_blocking(move || {
```

Comment:
Remind me why we have to use the `spawn_blocking` variants?

Reply:
We always call `spawn_blocking` before we call `Self::x224_lock`, so that we don't block a worker thread.

The distinction between worker threads and blocking threads is a subtlety of Tokio where the analogy to of Tokio <--> Goroutine runtime breaks down. As I understand it, Tokio gives you lower level control (and therefore finer grained control over performance) by exposing this distinction, whereas Go's runtime just takes care of any blocking/non-blocking distinction for you, at the expense of control over how threads are orchestrated under the hood.

See a good intro to worker vs blocking threads [here](https://github.com/tokio-rs/tokio/discussions/3858#discussioncomment-869878)